docs/installation-guide.md

```diff
@@ -0,0 +1,67 @@
+```
+npm -v
+```
+```
+node -v
+```
+
+If you don't have either Node or Npm installed, do so using the [official node.js installer](https://nodejs.org/en/download/).
+
+If you have the right versions installed, proceed to the CLI installation guide below else, upgrade the version of npm or node if lower than the recommended versions specified above.
+
+## Using the AsyncAPI Generator CLI
+The CLI tool allows you to generate message-based API boilerplate code, documentation or anything else you need as long as you specify it in your [template](template.md). To use the Generator via the CLI, you will need to install the AsyncApi Generator CLI tool in your local development environment.
+
+### Installation
+
+The recommended method to interact with the generator tool is by using its CLI. To use the CLI, install it  using the following `npm` command:
+```
+npm install -g @asyncapi/generator
+```
+
+To install a specific version of the generator tool, pass the version during installation:
+```
+npm install -g @asyncapi/generator@{version}
+```
+> :bulb: **Remember:** 
+> Each [community developed template](https://github.com/search?q=topic%3Aasyncapi+topic%3Agenerator+topic%3Atemplate) is dependent on a certain version of the generator for it to work correctly. So before you install the generator CLI, check the templates' `package.json` for the version of the Generator CLI your template is compatible with. To learn more about the generator version and how it correlates to the template you use check the [versioning docs](versioning.md).
+
+### Updating the AsyncAPI Generator CLI
+There are several reasons why you might want to update your generator version:
+1. You have the generator tool installed but you want to use the latest released features. To upgrade to the latest version, use the command below:
+```
+npm install -g @asyncapi/generator
+```
+2. Your template might not be compatible with the latest generator version so you update to a specific version of the generator. Check the [version you need](https://github.com/asyncapi/generator/releases) and specify the version you want to use using the **@** symbol as shown in the command below:
+```
+npm install -g @asyncapi/generator@{version}
+```
+> Sometimes you have to force additional npm installation like this: `npm install -g --force @asyncapi/generator`
+
+### Uninstall the AsyncAPI Generator CLI
+To uninstall the generator, use the following command:
+```
+npm uninstall @asyncapi/generator -gnpm uninstall @asyncapi/generator -g
+``` 
+
+> :memo: **Note:**  If you want to use the Generator CLI in your CI/CD pipeline to automate whatever you want generated for your event driven architecture apps, install the AsyncAPI Generator CLI in your pipeline. Incase you are using Github Actions, use [Github Actions for Generator](https://github.com/marketplace/actions/generator-for-asyncapi-documents).
+
+## Use the Generator Library In Your Node.js Apps
```

Comment:
```suggestion
## Generator Library in Node.js Apps
```

This is about how the URL and your page outline make sense and seem cleaner this way:
`asyncapi.com/docs/tools/generator#generator-library-in-nodejs-apps`